src/js/game/hud/parts/shape_tooltip.js

```diff
@@ -0,0 +1,89 @@
+import { DrawParameters } from "../../../core/draw_parameters";
+import { Vector } from "../../../core/vector";
+import { Entity } from "../../entity";
+import { KEYMAPPINGS } from "../../key_action_mapper";
+import { BaseHUDPart } from "../base_hud_part";
+
+export class HUDShapeTooltip extends BaseHUDPart {
+    createElements(parent) {}
+
+    initialize() {
+        /** @type {Vector} */
+        this.currentTile = new Vector(0, 0);
+
+        /** @type {Entity} */
+        this.currentEntity = null;
+
+        this.isPlacingBuilding = false;
+        this.active = false;
+
+        this.root.signals.entityQueuedForDestroy.add(() => {
+            this.currentEntity = null;
+        }, this);
+
+        this.root.hud.signals.selectedPlacementBuildingChanged.add(metaBuilding => {
+            this.isPlacingBuilding = metaBuilding;
+        }, this);
+    }
+
+    update() {
```

Comment:
Why have an update() and an active() method? Just make the update() method a isActive() method and call it in the draw() method. Or am I missing something?

Reply:
Then you also don't need to store an active and currentEntity flag

Reply:
Let me know if you want me to adjust it myself :p

Reply:
yeah that makes sense. And then update isn't even needed, it can all be in the draw method. Will do that now

Reply:
Okay that's done, I could also just put the isActive checks in draw, but it may just be simpler to leave it